### ISSUE TITLE
canvas: Create a formal ODA-Canvas release packaged in a Helm Chart repository

### ISSUE THREAD

**Issue by LesterThomas** [tmforum-oda/oda-canvas#160]
## Description

The current Canvas installation requires cloning the oda-canvas repository and installing using helm. At present, we don't use the versioning capability within helm to create a formal release (by following the installation instructions, you will install the latest canvas charts published in the master branch.)

### Proposal

Use the helm chart versioning to create identifiable versions, and publish these in a Helm Chart repository (so that you don't need to clone the repo to install, and to make traceable, tested version to help with troubleshooting). There are instructions to create a chart repository at https://helm.sh/docs/topics/chart_repository/ and we have created a chart repository for the reference example components at  https://github.com/tmforum-oda/reference-example-components 

The reference-example-components  uses a GitHub action to package each release into the chart repository, which is hosted using GitHub pages. You can see the released components at https://github.com/tmforum-oda/reference-example-components/releases

The helm chart versioning uses semantic versioning for the chart, and also allows a separate versioning for the 'app'. I propose that the  'app' versioning represents the latest version of the ODA-Component standard that the Canvas supports (currently v1beta3). I also propose that we use the semantic versioning of the chart and change the major version when the underlying component standard changes (e.g. we move from v1.x.x or the chart to v2.x.x when we move from v1beta3 to v1beta4). We update the minor version when we add new capability (e.g. the canvas supports a new option for some supporting service) and use the patch for bug-fixes.

As a separate follow-up issue, we should introduce more formal automated testing for each release (with build status badges as appropriate).

**Comment by brian-burton** [comment]
The issue raised in #164 means that we need to make a fairly rapid breaking change. To protect our community members from the breaking change we do need to implement formal releases now, so I propose the following:

- Create a Release to formalise the status of the Canvas as-was before we started version controlling it. This will allow existing work to continue without disruption.
- Merge changes from this issue, #164 and #165.
- Create a new Release to mark the change to following new release name/number conventions.

**Comment by LesterThomas** [comment]
I have done some experimentation around this. The proposal is to create a release 0.1.0 with the current (legacy) canvas, and then immediately create a release 1.0.0 incorporating the CamelCase CRD (as proposed by @brian-burton ). Here are the proposed steps to make this change:

In the oda-canvas repository:
* Change `master` branch name to `main`
* Create an empty gh-pages branch (which will just include a .nojekyll file).
* Update the settings to publish a GitHub pages from the gh-pages branch

Then in a separate organisation (so that we can fully test the GitHub actions and helm chart repo before issuing a Pull Request to merge).

* Create a Fork.
* In the forked repository, create an empty gh-pages branch (which will just include a .nojekyll file).
* Move the helm charts out of /installation and into a /charts folder.
* Update the README in the /installation to describe how to install from the Helm Chart Repo (instead of cloning the Repo).
* Create a GitHub pages in the repository settings.
* Create the GitHub actions release.yaml file to trigger the release.
* Change the versions of all the canvas helm charts to 0.1.0 and update the chart files to have a meaningful description.
* Commit changes to the fork, and check the GitHub actions are triggered and work successfully.
* Resolve any issues in the GitHub actions.

Then, issue a Pull-Request to merge this back into `tmforum-oda/oda-canvas` and test that the legacy release works.

Finally, update the helm charts to v1.0.0 and merge the `https://github.com/tmforum-oda/oda-canvas/tree/164-crd-kind-definition-should-be-camelcase` branch that fixes the CamelCase issue.
